Userland/Libraries/LibC/string.cpp

```diff
@@ -354,6 +354,23 @@ const char* const sys_errlist[] = {
 
 int sys_nerr = EMAXERRNO;
 
+int strerror_r(int errnum, char* buf, size_t buflen)
+{
+    if (errno >= EMAXERRNO)
+        return EINVAL;
```

Comment:
Dr. Posix encourages still copying an error string into the user's buffer even if we return EINVAL. Which is kinda weird but probably has a good reason. 

> Similarly, implementations are encouraged to have strerror_r() return [EINVAL] and put a string like "unknownerror" or "errornumberxxx" in the buffer pointed to by strerrbuf when the value of errnum is not a valid error number.

Reply:
Aye, that makes sense. I've changed that and added some tests as well. :)